### PR TITLE
Add download option to skip accession numbers

### PR DESCRIPTION
**Pull request by spolcyn**
This improves download efficiency when some filings have been already downloaded to another location, such as when doing incremental updates.

The existing logic to skip already downloaded files when they are in the same `download_folder` is useful for local-only use, but is less useful when files are only local for a brief period before being sent to long-term storage.

### CODE REVIEW

**Comment by spolcyn** [comment]
@jadchaar This feature has been manually tested -- would love a quick sanity check that this feature is reasonable to add and that I didn't miss it somewhere. Then will add automated tests.

File: sec_edgar_downloader/_Downloader.py [hunk 1]
```diff
@@ -67,6 +67,7 @@ def get(
         before: Optional[Date] = None,
         include_amends: bool = False,
         download_details: bool = False,
+        skip_accession_numbers: Optional[set[str]] = None,
```
Comment (jadchaar) [inline sec_edgar_downloader/_Downloader.py:70]: Please use Set from typing library here :)

File: sec_edgar_downloader/_Downloader.py [hunk 2]
```diff
@@ -173,6 +175,7 @@ def get(
                 download_details,
                 # Save ticker if passed in to form file system path for saving filings
                 ticker=ticker_or_cik if not is_cik(ticker_or_cik) else None,
+                skip_accession_numbers=skip_accession_numbers,
```
Comment (jadchaar) [inline sec_edgar_downloader/_Downloader.py:178]: This makes it seem like the value is a boolean, maybe rename to something like `accession_numbers_to_skip`

File: sec_edgar_downloader/_Downloader.py (hunk 1 above)
Comment (jadchaar) [reply sec_edgar_downloader/_Downloader.py:70]: IMHO we should not expose this as a parameter, but instead automatically use the existing file system to determine which items to skip. 

Thoughts?

**Review by jadchaar — CHANGES_REQUESTED** [review]
Thanks for the contribution! Left some feedback and ran the test suite against your changes

**Comment by spolcyn** [comment]
I have applied the suggestions and added a test. Let me know if there are any outstanding issues!

**Comment by spolcyn** [comment]
@jadchaar Any further issues or clarifications with this PR?

**Comment by jadchaar** [comment]
Thanks again for the contribution @spolcyn. LGTM!

**Review by jadchaar — APPROVED** [review]
LGTM! Ship it!

**Comment by Elijas** [comment]
@jadchaar This hasn't been uploaded to pypi

The latest version is 5.0.2 which was released at 2023-10-08


## Proof

<details>
  <summary>Source</summary>
  
```python
import requests
import zipfile
import tarfile
import io

def search_in_package(package: str, version: str, search_string: str) -> None:
    # Get package JSON metadata
    url = f"https://pypi.org/pypi/{package}/{version}/json"
    response = requests.get(url)
    data = response.json()

    # Find a suitable file to download (prefer wheel over sdist)
    download_url = None
    for file_info in data.get("urls", []):
        if file_info.get("packagetype") == "bdist_wheel":
            download_url = file_info.get("url")
            break
    if not download_url:
        # fallback to source distribution if wheel not found
        for file_info in data.get("urls", []):
            if file_info.get("packagetype") == "sdist":
                download_url = file_info.get("url")
                break

    if not download_url:
        raise ValueError("No downloadable file found for the package version.")

    print("Downloading from:", download_url)
    r = requests.get(download_url)
    r.raise_for_status()

    found = False

    # Helper function to search content for the target string
    def search_content(content, filename):
        nonlocal found
        if search_string in content:
            print(f"Found '{search_string}' in file: {filename}")
            found = True

    # Process wheel files (zip archives)
    if download_url.endswith(".whl"):
        with zipfile.ZipFile(io.BytesIO(r.content)) as z:
            for name in z.namelist():
                try:
                    with z.open(name) as f:
                        content = f.read().decode("utf-8", errors="ignore")
                        search_content(content, name)
                except Exception as e:
                    # If a file can’t be read as text, skip it
                    continue

    # Process source distributions (tar.gz archives)
    elif download_url.endswith((".tar.gz", ".tgz")):
        with tarfile.open(fileobj=io.BytesIO(r.content), mode="r:gz") as tar:
            for member in tar.getmembers():
                if member.isfile():
                    try:
                        f = tar.extractfile(member)
                        if f:
                            content = f.read().decode("utf-8", errors="ignore")
                            search_content(content, member.name)
                    except Exception as e:
                        continue
    else:
        print("Unknown file format.")

    if not found:
        print(f"'{search_string}' was not found in any file.")
  ```
  
</details>

https://github.com/jadchaar/sec-edgar-downloader/blob/15ebd0d538cd28145dffc87b5b6b88bcc579b655/sec_edgar_downloader/_Downloader.py#L69

```python
search = "download_details: bool = False"
search_in_package("sec-edgar-downloader", "5.0.2", search)
```
```
Found 'download_details: bool = False' in file: sec_edgar_downloader/_Downloader.py
Found 'download_details: bool = False' in file: sec_edgar_downloader/_types.py
```

https://github.com/jadchaar/sec-edgar-downloader/blob/15ebd0d538cd28145dffc87b5b6b88bcc579b655/sec_edgar_downloader/_Downloader.py#L70

```python
search = "accession_numbers_to_skip"
search_in_package("sec-edgar-downloader", "5.0.2", search)
```
```
'accession_numbers_to_skip' was not found in any file.
```

Related https://github.com/jadchaar/sec-edgar-downloader/issues/146

**Comment by jadchaar** [comment]
Ah thanks @Elijas, I will publish it shortly!

**Pull request merged**